src/System.Text.Json/tests/Serialization/OptionsTests.cs

```diff
@@ -321,5 +322,132 @@ public static void JsonEncodedTextStringsCustomAllowAll(string message, string e
             obj = JsonSerializer.Deserialize<TestClassForEncoding>(json);
             Assert.Equal(obj.MyString, message);
         }
+
+        [Fact]
+        public static void Options_GetConverterForObjectJsonElement_GivesCorrectConverter()
+        {
+            GenericObjectOrJsonElementConverterTestHelper<object>("JsonConverterObject", new object(), "[3]", true);
+            JsonElement element = JsonDocument.Parse("[3]").RootElement;
+            GenericObjectOrJsonElementConverterTestHelper<JsonElement>("JsonConverterJsonElement", element, "[3]", false);
+        }
+
+        private static void GenericObjectOrJsonElementConverterTestHelper<T>(string converterName, object objectValue, string stringValue, bool throws)
+        {
+            var options = new JsonSerializerOptions();
+
+            JsonConverter<T> converter = (JsonConverter<T>)options.GetConverter(typeof(T));
+            Assert.Equal(converterName, converter.GetType().Name);
+            
+            ReadOnlySpan<byte> data = Encoding.UTF8.GetBytes(stringValue);
+            Utf8JsonReader reader = new Utf8JsonReader(data);
+            reader.Read();
+            T readValue = converter.Read(ref reader, typeof(T), null);
+
+            if (readValue is JsonElement element)
```

Comment:
nit: I think it would be cleaner to just assert that reader is JsonElement (rather than doing `Assert.True(false)` in the else branch).

```suggestion
            Assert.True(readValue is JsonElement);
            var element = (JsonElement)readValue;
```

Reply:
I think i can do `Assert.True(readValue is JsonElement element);`, will apply with other PR as this one already merged

Reply:
Cannot be within assert because of scope, `(JsonElement)readValue;` complaining `cannot convert type T to JsonElement`. Need to keep if statement